src/libshadertrap/src/executor.cc

```diff
@@ -114,7 +115,36 @@ bool Executor::VisitAssertEqual(CommandAssertEqual* assert_equal) {
   if (assert_equal->GetArgumentsAreRenderbuffers()) {
     return CheckEqualRenderbuffers(assert_equal);
   }
-  return CheckEqualBuffers(assert_equal);
+  else{
+    if(assert_equal->GetFormatEntries().size() != 0){
+      for (const auto& format_entry : assert_equal->GetFormatEntries()) {
+        switch (format_entry.kind) {
+          case CommandAssertEqual::FormatEntry::Kind::kSkip:
+            return true;
+            break;
+          case CommandAssertEqual::FormatEntry::Kind::kString:
+            return true;
+            break;
+          case CommandAssertEqual::FormatEntry::Kind::kByte:
+            return CheckEqualBuffers<uint8_t>(assert_equal);
+            break;
+          case CommandAssertEqual::FormatEntry::Kind::kInt:
+            return CheckEqualBuffers<int32_t>(assert_equal);
+            break;
+          case CommandAssertEqual::FormatEntry::Kind::kUint:
+            return CheckEqualBuffers<uint32_t>(assert_equal);
+            break;
+          case CommandAssertEqual::FormatEntry::Kind::kFloat:
+            return CheckEqualBuffers<float>(assert_equal);
+            break;
```

Comment:
It appears that this will check the whole buffer for equality according to the given data type for every different format entry.

That's not what we want. Instead, we want to get all of the buffer data, and then process the data in chunks according to the types in the format entries.

I suggest doing all this work inside CheckEqualBuffers. I.e., revert this method, remove the template argument from CheckEqualBuffers, and then adjust CheckEqualBuffers so that it uses format entry information.